ios/ReactNativePageView.m

```diff
@@ -10,412 +9,333 @@
-
-@property(nonatomic, strong) NSHashTable<UIViewController *> *cachedControllers;
-@property (nonatomic, assign) CGPoint lastContentOffset;
-
-- (void)goTo:(NSInteger)index animated:(BOOL)animated;
-- (void)shouldScroll:(BOOL)scrollEnabled;
-- (void)shouldShowPageIndicator:(BOOL)showPageIndicator;
-- (void)shouldDismissKeyboard:(NSString *)dismissKeyboard;
-
 
 @end
 
-@implementation ReactNativePageView {
-    uint16_t _coalescingKey;
-}
 
-- (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher {
-    if (self = [super init]) {
-        _scrollEnabled = YES;
-        _pageMargin = 0;
-        _previousIndex = -1;
-        _transitionStyle = UIPageViewControllerTransitionStyleScroll;
-        _orientation = UIPageViewControllerNavigationOrientationHorizontal;
-        _currentIndex = 0;
-        _dismissKeyboard = UIScrollViewKeyboardDismissModeNone;
-        _coalescingKey = 0;
-        _eventDispatcher = eventDispatcher;
-        _cachedControllers = [NSHashTable weakObjectsHashTable];
-        _overdrag = YES;
-    }
-    return self;
-}
+@implementation ReactNativePageView
 
-- (void)layoutSubviews {
-    [super layoutSubviews];
-    if (self.reactPageViewController) {
-        [self shouldScroll:self.scrollEnabled];
-        //Below line fix bug, where the view does not update after orientation changed.
-        [self updateDataSource];
-    }
-}
-
-- (void)didUpdateReactSubviews {
-    if (!self.reactPageViewController && self.reactViewController != nil) {
+- (instancetype)init {
+    if (self = [super init]) {
+        self.controllerCache = [NSMapTable weakToWeakObjectsMapTable];
+        self.currentPage = 0;
+        self.currentReactTag = nil;
+        self.orientation = UIPageViewControllerNavigationOrientationHorizontal;
+        self.overdrag = YES;
+        self.pageMargin = 0;
+        self.placeholderPageIndexes = [NSMapTable weakToStrongObjectsMapTable];
+        self.scrollEnabled = YES;
+        self.showPageIndicator = NO;
+        self.transitionStyle = UIPageViewControllerTransitionStyleScroll;
         [self embed];
-        [self setupInitialController];
+        return self;
     } else {
-        [self updateDataSource];
+        return nil;
     }
 }
 
-- (void)didMoveToSuperview {
-    [super didMoveToSuperview];
-    if (!self.reactPageViewController && self.reactViewController != nil) {
-        [self embed];
-        [self setupInitialController];
+- (void)didUpdateReactSubviews {
+    NSInteger targetPage = self.currentPage;
+    NSInteger reactIndex = [self getReactIndexOfView:self.currentReactTag];
+    if (reactIndex != NSNotFound) {
+        targetPage = self.offset + reactIndex;
     }
+    [self goTo:targetPage animated:NO];
 }
 
-- (void)didMoveToWindow {
-    [super didMoveToWindow];
-    if (!self.reactPageViewController && self.reactViewController != nil) {
+- (void)didSetProps:(NSArray<NSString *> *)changedProps {
+    if (
+        [changedProps containsObject:@"orientation"]
```

Comment:
why additional logic regarding to props setting is added ?

Reply:
For switching rendered pages, the pager must do more maintenance on caches.  However, native props can change much more frequently in lazy mode, so prefer to do as little maintenance as possible.

This has the extra benefit of making it possible to dynamically switch orientation/transition without requiring unmounting the component.

Reply:
Thank you for an explanation